packages/types/src/session.ts

```diff
@@ -130,11 +130,29 @@ export interface SessionResource extends ClerkResource {
   __internal_toSnapshot: () => SessionJSONSnapshot;
 }
 
+/**
+ * Represents a session resource that has completed all pending tasks
+ * and authentication factors
+ */
 export interface ActiveSessionResource extends SessionResource {
   status: 'active';
   user: UserResource;
 }
 
+/**
+ * Represents a session resource that is authenticated but has pending tasks
+ */
+export interface PendingSessionResource extends SessionResource {
+  status: 'pending';
+  user: UserResource;
+}
+
+/**
+ * Represents session resources for users who have completed
+ * the full authentication flow.
+ */
+export type AuthenticatedSessionResource = ActiveSessionResource | PendingSessionResource;
```

Comment:
Thought was better to add some JSDocs explaining the actual meaning and difference between those statuses, feel free to leave more feedback on it!